src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -663,11 +668,60 @@ public boolean isWithinSolidBlock() {
     // Health
 
     @Override
-    public <T extends Projectile> T launchProjectile(Class<? extends T> projectile,
-        Vector velocity) {
-        T entity = world.spawn(getEyeLocation(), projectile);
-        entity.setVelocity(velocity);
-        return entity;
+    public <T extends Projectile> T launchProjectile(Class<? extends T> clazz, Vector vector) {
+        float offset = 0.0F;
+        float velocity = 1.5F;
```

Comment:
Rename velocity to speed here, and any where else velocity is being defined as a magnitude. Then rename vector to velocity.

Reply:
Done in all throwProjectile overloads.